src/qibolab/channel.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass, field
+
+from qibo.config import raise_error
+
+from .channel_configs import ChannelConfig
+
+
+@dataclass(frozen=True)
+class Channel:
+    """Channel is an abstract concept that defines means of communication
+    between users and a quantum computer.
+
+    A quantum computer can be perceived as just a set of channels where
+    signals can be sent to or received from. Channels are identified
+    with a unique name. The type of a channel is inferred from the type
+    of config it accepts.
+    """
+
+    name: str
+    config: ChannelConfig
+
+
+@dataclass
+class ChannelMap:
+    """Collection of :class:`qibolab.designs.channel.Channel` objects
+    identified by name.
+
+    Essentially, it allows creating a mapping of names to channels just
+    specifying the names.
+    """
+
+    _channels: dict[str, Channel] = field(default_factory=dict)
```

Comment:
I did this myself, and I don't remember exactly the reason why I chose composition over inheritance (I guess for the typing, but I'm not sure...).
But reading the present code, I believe we could go for inheritance:
```suggestion
class ChannelMap(dict[str, Channel]):
    """Collection of :class:`qibolab.designs.channel.Channel` objects
    identified by name.

    Essentially, it allows creating a mapping of names to channels just
    specifying the names.
    """
```
(or just `dict`, if type hints are not interpreted as bare classes in inheritance - and then find a proper way to give it a type, or give up if unreasonable)

This would save the `__getitem__` and `__contains__` implementation. And I would even avoid the runtime check over the value class in `__setitem__`, and just rely on the static check (and the developers).

Reply:
I would even consider dropping this completely. Does it provide that much functionality over a `dict`? I believe it was added as a helper for the platform constructors, so that will depend on the template we will propose for that.

If we keep, maybe subclassing `dict` could simplify it a bit.

Reply:
I have been thinking about dropping it as well. Have not made my mind yet. As far as I understand the only benefit that this class has over just a dict is the following. With dict, adding new channel will look like `channel_map["my_qubits.drive.channel"] = Channel(name="my_qubits.drive.channel", config=...)`, while with this class you can do `channel_config.add(Channel(name="my_qubits.drive.channel", config=...))`, thus not repeating the name.

The channel maps are defined in various platform definitions, I suppose that's why this class was introduced to make the life's of people writing platform definitions easier, but I believe this can be avoided entirely by just constructing and handling a map inside Platform itself.

Reply:
This class does not exist anymore.